internal/caching/set.go

```diff
@@ -66,7 +76,25 @@ func (ss *Set) ContainsOrAdd(v []byte) bool {
 		ss.flop, ss.flip = ss.flip, ss.flop
 		log.Debugw("Cleared flop and swapped subsets as max size is reached", "maxSize", ss.maxSize)
 	}
-	return false
+	return false, nil
+}
+
+func (ss *Set) newKey(namespace, v []byte) ([32]byte, error) {
+	hasher, err := blake2b.New(blake2b.Size256, namespace)
+	if err != nil {
+		return [32]byte{}, err
+	}
+	_, err = hasher.Write(v)
+	if err != nil {
+		return [32]byte{}, err
+	}
+	var key [32]byte
+	keyB := hasher.Sum(key[:0])
+	copy(key[:], keyB)
+	// TODO: using blake's built-in keys means we lose the nice [32]byte return types
+	//       from Sum256. Consider pooling byte slices or other .Sum trickery to reduce
+	//       allocations.
```

Comment:
Well, we are doing "trickery" here. But, `key` _does_ get moved to the stack, unfortunately.